src/movement.cpp

```diff
@@ -643,14 +645,14 @@ uint32_t MoveEvent::StepInField(Creature* creature, Item* item, const Position&)
 	return LUA_ERROR_ITEM_NOT_FOUND;
 }
 
-uint32_t MoveEvent::StepOutField(Creature*, Item*, const Position&) { return 1; }
+uint32_t MoveEvent::StepOutField(std::shared_ptr<Creature>, std::shared_ptr<Item>, const Position&) { return 1; }
 
-uint32_t MoveEvent::AddItemField(Item* item, Item*, const Position&)
+uint32_t MoveEvent::AddItemField(std::shared_ptr<Item> item, std::shared_ptr<Item>, const Position&)
 {
-	if (MagicField* field = item->getMagicField()) {
-		Tile* tile = item->getTile();
+	if (auto field = item->getMagicField()) {
```

Comment:
Could this be `const auto`
```suggestion
	if (const auto field = item->getMagicField()) {
```